src/classes/EP_ManageEPTemplate_CTRL.cls

```diff
@@ -0,0 +1,629 @@
+        buttonGroup.childComponents.add(deleteTaskButton);
+        cardFooter.childComponents.add(buttonGroup);
+        cardFooter.childComponents.add(commentsInputPanel(wrapper, accessString, 'Comments__c', 'slds-col slds-p-left--medium', 'slds-textarea'));
+
+        topLevelPanel.childComponents.add(formPanel);
+        formPanel.childComponents.add(boxAndLabel);
+        
+        formPanel.childComponents.add(firstRow);
+        formPanel.childComponents.add(secondRow);
+        formPanel.childComponents.add(cardFooter);
+
+        //closes the box
+        formPanel.childComponents.add(new Component.Apex.OutputText(value='</fieldset></div>', escape=false));
+
+        //recursively call this method to add children
+        for (integer i = 0; i < wrapper.children.size(); i++) {
+            topLevelPanel.childComponents.add( addChildComponents(wrapper.children[i], i, accessString + '.children', levelString+'-'+(i+1)) );
+        } 
+        return topLevelPanel;
+    }
+
+    /** @description Generates the markup for an SLDS styled form element with label.
+    * @param wrapper The TaskWrapper for the current Engagement Plan Task.
+    * @param fieldName The name of the field.
+    * @param outterCss CSS markup for the field input container.
+    * @return Component.Apex.OutputPanel A form field container and label.
+    */
+    private Component.Apex.OutputPanel fieldLabelPanel(TaskWrapper wrapper, String fieldName, String outterCss) {
+        Component.Apex.OutputPanel result = new Component.Apex.OutputPanel(styleClass='slds-form-element ' + outterCss, layout='block');
+        Component.Apex.OutputLabel fieldLabel = new Component.Apex.OutputLabel(
+            value=UTIL_Describe.getFieldLabel(
+                UTIL_Namespace.StrTokenNSPrefix('Engagement_Plan_Task__c'),
+                //don't namespace prefix standard fields like Name
+                fieldName.endsWith('__c')?UTIL_Namespace.StrTokenNSPrefix(fieldName):fieldName
+            ), 
+            styleClass='slds-form-element__label'
+        );
+        fieldLabel.for = fieldName + wrapper.level;
+        result.childComponents.add(fieldLabel);
+        return result;
+    }
+
+    /** @description Generates the markup for an SLDS styled apex:inputField.
+    * @param wrapper The TaskWrapper for the current Engagement Plan Task.
+    * @param accessString The string for accessing the current place in the heirarchy using a nested expression.
+    * @param fieldName The name of the field.
+    * @param outterCss CSS markup for the field input container.
+    * @param inputCss CSS markup for the input itself.
+    * @return Component.Apex.OutputPanel OutputPanel containing the markup for an inputField.
+    */
+    private Component.Apex.OutputPanel genericInputPanel(TaskWrapper wrapper, String accessString, String fieldName, String outterCss, String inputCss) {
+        Component.Apex.OutputPanel outterPanel = fieldLabelPanel(wrapper, fieldName, outterCss);
```

Comment:
for required fields (ie, name), ideally you would add a preceding red asterisk, to mimic how Lightning Experience displays required fields.